src/analysis/modals/ComputeModal/PersistentDiskTypeInput.ts

```diff
@@ -1,30 +1,30 @@
-import { useUniqueId } from '@terra-ui-packages/components';
-import React from 'react';
+import { Select, SelectProps, useUniqueId } from '@terra-ui-packages/components';
+import { ReactNode } from 'react';
 import { div, h, label } from 'react-hyperscript-helpers';
 import { IComputeConfig } from 'src/analysis/modal-utils';
 import { computeStyles } from 'src/analysis/modals/modalStyles';
-import { Select } from 'src/components/common';
-import { PdSelectOption, SharedPdType } from 'src/libs/ajax/leonardo/models/disk-models';
+import { DiskType } from 'src/libs/ajax/leonardo/models/disk-models';
 
-export interface PersistentDiskTypeInputProps {
-  value: SharedPdType;
-  onChange: (e: { value: SharedPdType; label: string | undefined } | null) => void;
-  isDisabled?: boolean;
-  options: PdSelectOption[];
-}
+type PersistentDiskTypeSelectProps<
+  T extends IComputeConfig['persistentDiskType'],
+  Option extends { value: T; label: string }
+> = SelectProps<T, false, Option>;
```

Comment:
Using a generic here allows for value, options, and onChange to be typed with either `AzureDiskType` or `GoogleDiskType` in the Azure/Google compute modals instead of the current situation where both of those are typed with `DiskType`.